microsoft-edge/webview2/concepts/process-related-events.md

```diff
@@ -0,0 +1,636 @@
+
+* **Unexpected exit.** The process indicated by the event has exited unexpectedly (usually due to a crash). The failure might or might not be recoverable and some failures are auto-recoverable. See [Handle unexpected exits for various types of processes](#handle-unexpected-exits-for-various-types-of-processes) below for details about which of these can be handled by your application.  When the impacted process is the main browser process, a `BrowserProcessExited` event will be raised too; see [Handle main browser process crashes](#handle-main-browser-process-crashes) below.
+
+* **Unresponsiveness.** A **renderer process** might become unresponsive to user input.  This is only reported for renderer processes; see [Handle unresponsive renderers](#handle-unresponsive-renderers) below.
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+* [CoreWebView2.ProcessFailed Event](https://learn.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2.processfailed)
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+* [CoreWebView2.ProcessFailed Event](https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#processfailed)
+
+##### [Win32/C++](#tab/win32cpp)
+
+* [ICoreWebView2::add_ProcessFailed event](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2#add_processfailed)
+* [ICoreWebView2::remove_ProcessFailed event](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2#remove_processfailed)
+
+---
+
+These situations can be identified through the details provided in the event. See the section below for how your application can do this.
+
+Subsections below:
+* [Gather process failure details](#gather-process-failure-details)
+* [Handle unexpected exits for various types of processes](#handle-unexpected-exits-for-various-types-of-processes)
+* [Handle unresponsive renderers](#handle-unresponsive-renderers)
+
+
+<!-- ------------------------------ -->
+#### Gather process failure details
+
+
+The `ProcessFailed` event provides information about the **kind of failure** and the **reason** why it occurred. Your application can interpret these details in the following manner.
+
+Subsections below:
+* [Failure kind](#failure-kind)
+* [Failure reason](#failure-reason)
+* [Exit code, process description, and frames information](#exit-code-process-description-and-frames-information)
+
+
+<!-- ---------- -->
+###### Failure kind
+
+Due to historical reasons, _failure kind_ is a combination of process kind (browser, renderer, gpu, ...) and failure (exit, unresponsiveness). Your application can use _failure kind_ to determine the process that has failed. Renderer processes are further divided in _main frame_ renderer (`RenderProcessExited`, `RenderProcessUnresponsive`) and _subframe_ renderer (`FrameRenderProcessExited`). For more details about the conditions under which each specific _failure kind_ is used, see:
```

Comment:
Is there a reason we're using _failure kind_, _failure reason_, etc. instead of `ProcessFailedKind`, `ProcessFailedReason`, etc.?

Reply:
Just that it is more of a conceptual term than a property name. I don't think it makes much of a difference. Updated.